src/main/java/org/primefaces/component/fileupload/FileUploadRenderer.java

```diff
@@ -151,6 +158,9 @@ protected void encodeAdvancedMarkup(FacesContext context, FileUpload fileUpload)
 
         //buttonbar
         writer.startElement("div", fileUpload);
+        if (fileUpload.hasDropZoneFacet()) {
+            writer.writeAttribute("style", "display:none", null);
+        }
```

Comment:
Not too sure if I like this. It does prevent needing to change any JavaScript... 🤔